osu.Game/Rulesets/Mods/ModUsage.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Rulesets.Mods
+{
+    /// <summary>
+    /// The usage of this mod to determine its playability.
+    /// </summary>
+    public enum ModUsage
```

Comment:
Unsure how to properly document this mod, right now it feels completely off from its purpose (especially for the fact that `Solo` is meant to be for cases where multiplayer is irrelevant, not specifically a solo gameplay session).

Reply:
Maybe "user" is fine then. Dunno, I wasn't super sure I was going to get buy in from everyone on this idea so pushing it in here already seems rushed.

Reply:
I think it is a no-brainer to simplify from three cascading properties to one central method, but maybe. I've renamed `Solo` to `User` and adjusted the documentation to hopefully read better.